Sources/ImperialShopify/ShopifyRouter.swift

```diff
@@ -1,97 +1,81 @@
 import Vapor
 
-public class ShopifyRouter: FederatedServiceRouter {
-    
-    public let tokens: FederatedServiceTokens
-    public let callbackCompletion: (Request, String) throws -> (EventLoopFuture<ResponseEncodable>)
-    public var scope: [String] = []
+final public class ShopifyRouter: FederatedServiceRouter {
+    public let tokens: any FederatedServiceTokens
+    public let callbackCompletion: @Sendable (Request, String) async throws -> any AsyncResponseEncodable
+    public let scope: [String]
     public let callbackURL: String
-    public var accessTokenURL: String = ""
+    // `accessTokenURL` used to be set inside `authURL` and read by `fetchToken`
+    // now `fetchToken` creates the `accessTokenURL` itself from the shop domain in the request
+    public let accessTokenURL: String = ""
```

Comment:
Can you add a comment here that this is required by the protocol to make it clear why we have it